Code/Framework/AzToolsFramework/Tests/Metadata/UuidUtilTests.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/RTTI/ReflectContext.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzToolsFramework/Metadata/MetadataManager.h>
+#include <AzCore/Serialization/Json/JsonSystemComponent.h>
+#include <AzCore/Serialization/Json/RegistrationContext.h>
+#include <AzCore/UnitTest/Mocks/MockFileIOBase.h>
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzToolsFramework/Metadata/UuidUtils.h>
+
+namespace UnitTest
+{
+    struct UuidUtilTests
+        : LeakDetectionFixture
+        , AZ::ComponentApplicationBus::Handler
```

Comment:
could this be simpler if you used `UnitTest::MockComponentApplication` instead - like is used [here](https://github.com/o3de/o3de/blob/development/Code/Editor/Lib/Tests/test_AzAssetBrowserRequestHandler.cpp#L136)?

Reply:
I didn't know that existed, thanks for pointing that out, Ill update to use that instead